From 422569c3f7bb4b99797d6835125660475c053f35 Mon Sep 17 00:00:00 2001
From: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Date: Thu, 11 Aug 2022 15:04:12 +0530
Subject: [PATCH] ath12k: add MLO global memory support in coredump

Add MLO global memory type support in the coredump module. So that
SSR dump collect this memory region.

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/coredump.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/coredump.c b/drivers/net/wireless/ath/ath12k/coredump.c
index 0854502..94610ed 100644
--- a/drivers/net/wireless/ath/ath12k/coredump.c
+++ b/drivers/net/wireless/ath/ath12k/coredump.c
@@ -164,7 +164,8 @@ void ath12k_coredump_download_rddm(struct ath12k_base *ab)
 		if (ab->qmi.target_mem[i].type == HOST_DDR_REGION_TYPE ||
 		    ab->qmi.target_mem[i].type == CALDB_MEM_REGION_TYPE ||
 		    ab->qmi.target_mem[i].type == M3_DUMP_REGION_TYPE ||
-		    ab->qmi.target_mem[i].type == PAGEABLE_MEM_REGION_TYPE)
+		    ab->qmi.target_mem[i].type == PAGEABLE_MEM_REGION_TYPE ||
+		    ab->qmi.target_mem[i].type == MLO_GLOBAL_MEM_REGION_TYPE)
 			rem_seg_cnt++;
 	}
 
@@ -211,6 +212,15 @@ void ath12k_coredump_download_rddm(struct ath12k_base *ab)
 				continue;
 			}
 		}
+
+		if (ab->qmi.target_mem[i].type == MLO_GLOBAL_MEM_REGION_TYPE) {
+			seg_info->len = ab->qmi.target_mem[i].size;
+			seg_info->addr = ab->qmi.target_mem[i].paddr;
+			seg_info->vaddr = ab->qmi.target_mem[i].v.ioaddr;
+			seg_info->type = ATH12K_FW_MLO_GLOBAL;
+			seg_info++;
+			continue;
+		}
 		if (ab->qmi.target_mem[i].type == PAGEABLE_MEM_REGION_TYPE) {
 			seg_info->len = ab->qmi.target_mem[i].size;
 			seg_info->addr = ab->qmi.target_mem[i].paddr;
-- 
2.17.1

